src/java/org/apache/cassandra/cql3/functions/CollectionFcts.java

```diff
@@ -169,6 +208,22 @@ public ByteBuffer execute(ProtocolVersion protocolVersion, List<ByteBuffer> para
         };
     }
 
+    private static NativeScalarFunction makeCollectionCountFunction(String name, AbstractType<?> inputType)
+    {
+        return new NativeScalarFunction(name, Int32Type.instance, inputType)
+        {
+            @Override
+            public ByteBuffer execute(ProtocolVersion protocolVersion, List<ByteBuffer> parameters)
+            {
+                ByteBuffer value = parameters.get(0);
+                if (value == null)
+                    return null;
```

Comment:
Does it mean if counting `null`, it returns `null`, otherwise returns 1?

Reply:
Yes, it does. CQL functions applied to `null` generally return `null`, as in SQL. This function is only used for not-collection functions, where `collection_count` is either `null` or `1`. Right below there is the function overload for collections, that also returns `null` for `null` collections, and the collection size for not-null collections.